File: 1038/1038.c
int binarySearchTreeToGreaterSumTr(struct TreeNode* root) {
    if (!root) return 0;
    int l = binarySearchTreeToGreaterSumTr(root->left);
    int r = binarySearchTreeToGreaterSumTr(root->right);
    return 1 + (l > r ? l : r);
}

File: 1038/1038.cs
public class Solution {
    public int BinarySearchTreeToGreaterSumTr(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(BinarySearchTreeToGreaterSumTr(root.left), BinarySearchTreeToGreaterSumTr(root.right));
    }
}